packages/superset-ui-chart/src/clients/ChartClient.ts

```diff
@@ -94,7 +94,10 @@ export default class ChartClient {
           },
           ...options,
         } as RequestConfig)
-        .then(response => response.json as Json);
+        .then(response => {
+          // let's assume response.json always has the shape of QueryData
+          return response.json as QueryData;
```

Comment:
is this needed since the method return type is `Promise<QueryData>`?

Reply:
Yes, it is, because `response.json` might be `undefined`, then it becomes unassignable to `Promise<QueryData>`. Previously it was hinted as `Json`, I just changed it to `QueryData` to be more strict.